Mobile Buy SDK Sample Apps/Sample App Advanced/Mobile Buy SDK Advanced Sample/CheckoutViewController.m

```diff
@@ -195,7 +195,13 @@ - (void)showCheckoutConfirmation
 
 - (void)safariViewControllerDidFinish:(SFSafariViewController *)controller
 {
-    [self.navigationController popToRootViewControllerAnimated:YES];
+    [self getCompletedCheckout:^{
+        dispatch_async(dispatch_get_main_queue(), ^{
+            if (self.checkout.order) {
```

Comment:
reverse the order of the if-statement and dispatch_async so nothing is enqueued if order doesn't exist